elyra/cli/pipeline_app.py

```diff
@@ -63,22 +88,35 @@ def _preprocess_pipeline(pipeline_path, runtime, runtime_config):
     if len(pipeline_definition['pipelines']) == 0:
         raise click.ClickException("Pipeline has zero length 'pipelines' field.")
 
+    primary_pipeline_key = pipeline_definition['primary_pipeline']
+    primary_pipeline = None
     try:
         for pipeline in pipeline_definition["pipelines"]:
+            if pipeline['id'] == primary_pipeline_key:
+                primary_pipeline = pipeline
             for node in pipeline["nodes"]:
                 if 'filename' in node["app_data"]:
                     abs_path = os.path.join(pipeline_dir, node["app_data"]["filename"])
                     node["app_data"]["filename"] = abs_path
 
-        # NOTE: The frontend just set the info for first pipeline, but shouldn't it
-        # search for the primary pipeline and set that?
-        # Setting `pipeline_definition["pipelines"][0]` for consistency.
-        pipeline_definition["pipelines"][0]["app_data"]["name"] = pipeline_name
-        pipeline_definition["pipelines"][0]["app_data"]["runtime"] = runtime
-        pipeline_definition["pipelines"][0]["app_data"]["runtime-config"] = runtime_config
     except Exception as e:
         raise click.ClickException(f"Error pre-processing pipeline: \n {e}")
 
+    pipeline_version = int(primary_pipeline["app_data"]["version"])
+    if PIPELINE_CURRENT_VERSION != pipeline_version:
+        raise click.ClickException(f"Unsupported pipeline version {pipeline_version}. "
+                                   f"Version must be {PIPELINE_CURRENT_VERSION}.")
+
+    if not _validate_pipeline_runtime(primary_pipeline, runtime):
+        raise click.ClickException("Runtime mismatch - Trying to run a runtime specific pipeline"
+                                   " using a configuration for a different runtime.")
```

Comment:
This message should include user-friendly, actionable information. As is, we leave it up to the user to determine what needs to be done to resolve the problem. Keep in mind:
- the person using the CLI might not be the person who created the pipeline and therefore won't necessarily know of which type the pipeline is -> dev vs ops persona
- the person needs to either understand the structure of the `.pipeline` file to determine of which type it is or open the pipeline in the VPE

Reply:
How about something like this, with `{}` being placeholders for the appropriate values.

```
This pipeline requires a {Kubeflow Pipelines | Apache Airflow} runtime configuration. The specified configuration {rtc_name} is for {Kubeflow Pipelines | Apache Airflow}. 
```